### PR TITLE
Performance optimizations

### PR DESCRIPTION
**Pull request by sarveshnagpal**
- Reducing computation complexity of tokenization logic
- Simplifying fields in session cookie by removing backward compatibility code
- Honor prior consent setting to persist tracking option for the user

### CODE REVIEW

File: packages/clarity-js/src/data/metadata.ts [hunk 1]
```diff
@@ -162,6 +152,11 @@ function user(): User {
   let output: User = { id: shortid(), expiry: null };
   let cookie = getCookie(Constant.CookieKey);
   if(cookie && cookie.length > 0) {
+    // Presence of a valid first-party user cookie implies one of two things below:
+    //   - The project has tracking enabled
+    //   - We received prior consent for this user to enable tracking 
+    // In both cases we can assume tracking is enabled so we can attribute different page views to the same session.
+    config.track = true;
```
Comment (toby-walker) [inline packages/clarity-js/src/data/metadata.ts:159]: discuss offline.

File: packages/clarity-js/src/data/token.ts [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
     let reference = null;
-    for (let token of metadata) {
-        let index = tokens.indexOf(token);
-        if (index >= 0) {
-            if (reference) { reference.push(index); } else {
-                reference = [index];
-                tokens.push(reference);
+    for (let i = 0; i < tokens.length; i++) {
+        // Only optimize for string values
+        if (typeof tokens[i] === Constant.String) {
```
Comment (toby-walker) [inline packages/clarity-js/src/data/token.ts:16]: Suggestion: Add a unit test for this fn.

**Review by toby-walker — APPROVED** [review]
:shipit:

File: packages/clarity-js/src/data/metadata.ts (hunk 1 above)
Comment (sarveshnagpal) [reply packages/clarity-js/src/data/metadata.ts]: PR reflected to reflect our discussion.

**Pull request merged**